### PR TITLE
miscellaneous cleanup, comments and optimizations of ilsgateway migration code

### PR DESCRIPTION
**Pull request by czue**

### CODE REVIEW

**Comment by czue** [comment]
@kkrampa can you take a look through this and provide any comments you have? I put a "kkrampa" in the code where I thought you might know the answer, but feel free to provide feedback on any of the code or todo

**Comment by czue** [comment]
@kkrampa ping

File: custom/ilsgateway/api.py [hunk 1]
```diff
@@ -230,8 +233,11 @@ def location_sync(self, ilsgateway_location, fetch_groups=False):
                 location.longitude = float(ilsgateway_location.longitude)
             location.location_type = ilsgateway_location.type
             location.site_code = ilsgateway_location.code
+            # todo: unicode?
             location.external_id = str(ilsgateway_location.id)
             location.save()
+
+            # todo: shouldn't this only be creating supply points for objects just at the facility level?
```
Comment (kkrampa) [inline custom/ilsgateway/api.py]: @czue There are some sms users in ILS that are assigned to non-facility locations. In HQ when we assign location to user supply point is automatically created. That's reason why I'm creating supply point for all locations. Not sure how it should be solved.

File: custom/ilsgateway/api.py [hunk 2]
```diff
@@ -260,6 +267,8 @@ def location_sync(self, ilsgateway_location, fetch_groups=False):
             historical_groups = {}
             while counter != 5:
                 try:
+                    # todo: we may be able to avoid this call by passing the groups in as part of the original
```
Comment (kkrampa) [inline custom/ilsgateway/api.py]: @czue I think that fetching groups in chunk that contains 100 locations, always makes timeout.

File: custom/ilsgateway/tasks.py [hunk 3]
```diff
@@ -63,28 +63,39 @@ def get_locations(api_object, facilities):
...
+                                       'product_stock',
+                                       meta.get('limit') or limit,
+                                       meta.get('offset') or offset,
+                                       date, facility, True)
+            for product_stock in product_stocks:
+                # this logic updates the StockState object based on data from the current ProductStock
+                # todo: It seems a little bit odd/wrong that we would have this in the migration code at all
+                # Shouldn't we be able to rely on the stock transaction logic loading the right data
+                # into the StockState?
+
```
Comment (kkrampa) [inline custom/ilsgateway/tasks.py]: I think you're right that this is unnecessary. We don't calculate consumption during stock transaction but after that we should be able to generate StockState data

File: custom/ilsgateway/tasks.py [hunk 4]
```diff
@@ -106,40 +117,43 @@ def sync_product_stock(domain, endpoint, facility, checkpoint, date, limit=100,
...
+    if case:
+        while has_next:
+            meta, stocktransactions = endpoint.get_stocktransactions(next_url_params=next_url,
+                                                                     limit=limit,
+                                                                     offset=offset,
+                                                                     filters=(dict(supply_point=supply_point,
+                                                                                   date__gte=date,
+                                                                                   order_by='date')))
+
+            # todo: kkrampa, shouldn't we wait to save the checkpoint until after we've processed all the data?
```
Comment (kkrampa) [inline custom/ilsgateway/tasks.py]: It sets point where we should back if something goes wrong. So e.g. when offset=200 and something goes wrong, we get back next time and fetch data for offset=200 again. Basically, checkpoints store last processed data.

File: custom/ilsgateway/tasks.py [hunk 5]
```diff
@@ -200,6 +216,7 @@ def sync_supply_point_status(domain, endpoint, facility, checkpoint, date, limit
             filters=dict(supply_point=facility, status_date__gte=date),
             facility=facility
         )
+        # todo: kkrampa, shouldn't we wait to save the checkpoint until after we've processed all the data?
```
Comment (kkrampa) [inline custom/ilsgateway/tasks.py]: Same as above.

File: custom/ilsgateway/tasks.py [hunk 6]
```diff
@@ -231,6 +248,8 @@ def sync_delivery_group_report(domain, endpoint, facility, checkpoint, date, lim
                                                                          filters=dict(supply_point=facility,
                                                                                       report_date__gte=date),
                                                                          facility=facility)
+
+        # todo: kkrampa, shouldn't we wait to save the checkpoint until after we've processed all the data?
```
Comment (kkrampa) [inline custom/ilsgateway/tasks.py]: Same as above.

File: custom/logistics/tasks.py [hunk 7]
```diff
@@ -68,6 +70,7 @@ def stock_data_task(domain, endpoint, apis, test_facilities=None):
         )
         limit = 100
         offset = 0
+        # todo: kkrampa why is this here?
```
Comment (kkrampa) [inline custom/logistics/tasks.py]: Basically, It's caused by way that we are migrating data. First of all we are migrating all product stocks for all locations, then all stock transactions for all locations etc. Yeah, I know that probably It should be  flipped (get data location by location) but probably it doesn't speed up migration.

**Comment by kkrampa** [comment]
@czue I made some comments. Could I take a look on todos that you've made but not addressed to me or someone else will do it?

File: custom/ilsgateway/tasks.py (hunk 3 above)
Comment (kkrampa) [inline custom/ilsgateway/tasks.py]: Do you think that run this task: https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/commtrack/tasks.py#L19 should be enough to have valid stock state?

Comment (czue) [inline custom/ilsgateway/tasks.py]: @kkrampa not sure about that task exactly, but I think we can follow up on this later

File: custom/ilsgateway/tasks.py (hunk 4 above)
Comment (czue) [inline custom/ilsgateway/tasks.py]: got it. so just to confirm the offset that comes back from `meta` is the same offset that we pass in?

Comment (czue) [inline custom/ilsgateway/tasks.py]: (was able to tell from the API response that this is right)

**Comment by dannyroberts** [comment]
I couldn't follow too closely, but given your comments in HQ Devs skype chat and the fact that it appears to be mostly moving code around, looks fine to me. :+1: on all the comments and docstrings

File: custom/ilsgateway/tasks.py (hunk 3 above)
Comment (czue) [inline custom/ilsgateway/tasks.py]: @kkrampa https://github.com/dimagi/commcare-hq/pull/5907

**Pull request merged**